muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/communicator/rest/CommunicatorRESTService.java

```diff
@@ -1,912 +1,1059 @@
+  public Response listUserSentCommunicatorMessagesByMessageId( 
+      @PathParam ("COMMUNICATORMESSAGEID") Long communicatorMessageId) {
+    UserEntity user = sessionController.getLoggedUserEntity(); 
+    
+    CommunicatorMessageId threadId = communicatorController.findCommunicatorMessageId(communicatorMessageId);
+    
+    List<CommunicatorMessage> receivedItems = communicatorController.listMessagesByMessageId(user, threadId, false);
+
+    CommunicatorMessageId olderThread = communicatorController.findOlderThreadId(user, threadId, CommunicatorFolderType.SENT, null);
+    CommunicatorMessageId newerThread = communicatorController.findNewerThreadId(user, threadId, CommunicatorFolderType.SENT, null);
+    
+    List<CommunicatorMessageIdLabel> labels = communicatorController.listMessageIdLabelsByUserEntity(user, threadId);
+    List<CommunicatorMessageIdLabelRESTModel> restLabels = restModels.restLabel(labels);
+    
+    return Response.ok(
+      restModels.restThreadViewModel(receivedItems, olderThread, newerThread, restLabels)
+    ).build();
+  }
+  
+  @GET
+  @Path ("/searchItems/")
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response messageSearchFromInbox(
+      @QueryParam("message") String message,
+      @QueryParam("firstResult") @DefaultValue ("0") Integer firstResult, 
+      @QueryParam("maxResults") @DefaultValue ("10") Integer maxResults) {
+	
+    List<CommunicatorSearchResultRESTModel> communicatorMessages = new ArrayList<CommunicatorSearchResultRESTModel>();
+    
+    Iterator<SearchProvider> searchProviderIterator = searchProviders.iterator();
+    if (searchProviderIterator.hasNext()) {
+      SearchProvider searchProvider = searchProviderIterator.next();
+      SearchResults<List<IndexedCommunicatorMessage>> searchResult = null;
+      
+      List<Sort> sorts = null;
+      
+      if (message.isEmpty() || message == null) {
+    	 return Response.noContent().build();
+      }
+      
+      searchResult = searchProvider.searchCommunicatorMessages()
+          .setSorts(sorts)
+          .setMaxResults(maxResults)
+          .setFirstResult(firstResult)
+          .setMessage(message)
+          .search();
+      
+      try {
+        List<IndexedCommunicatorMessage> results = searchResult.getResults();
+        for (IndexedCommunicatorMessage result : results) {
+          Boolean readByReceiver = false;
+          
+          Long communicatorMessageId = result.getCommunicatorMessageId();
+          Long senderId = result.getSenderId();
+          
+          IndexedCommunicatorMessageSender sender = result.getSender();
+          Long recipientId = null;
+
+          Boolean isLogged = sender.getUserEntityId().equals(sessionController.getLoggedUserEntity().getId());
+          Boolean archivedBySender = Boolean.TRUE.equals(sender.getArchivedBySender());
+          if (isLogged.equals(Boolean.TRUE) && archivedBySender.equals(Boolean.TRUE)) {

```

Comment:
archived should probably be filtered in the search so that the indexing doesn't break